src/engine/engine_wa.c

```diff
@@ -24,7 +24,7 @@ static void _ocf_read_wa_complete(struct ocf_request *req, int error)
 
 	if (req->error) {
 		req->info.core_error = 1;
-		env_atomic_inc(&req->core->counters->core_errors.write);
+		ocf_core_stats_core_error_update(req, OCF_WRITE);
```

Comment:
Regarding all stat update functions taking ocf_request as parameter - I would consider passing the needed values directly to break the dependency between ocf_request and ocf_stats module